src/BalanceTx.purs

```diff
@@ -387,30 +380,20 @@ balanceTx unattachedTx@(UnattachedUnbalancedTx { unbalancedTx: t }) = do
     -- Get own wallet address, collateral and utxo set:
     ownAddr <- ExceptT $ getWalletAddress <#>
       note (GetWalletAddressError' CouldNotGetNamiWalletAddress)
-    collateral <- ExceptT $ getWalletCollateral <#>
-      note (GetWalletCollateralError' CouldNotGetNamiCollateral)
     utxos <- ExceptT $ utxosAt ownAddr <#>
       (note (UtxosAtError' CouldNotGetUtxos) >>> map unwrap)
-
     let
       -- Combines utxos at the user address and those from any scripts
       -- involved with the contract in the unbalanced transaction.
       allUtxos :: Utxo
       allUtxos = utxos `Map.union` utxoIndexToUtxo networkId utxoIndex
 
-      -- After adding collateral, we need to balance the inputs and
-      -- non-Ada outputs before looping, i.e. we need to add input fees
-      -- for the Ada only collateral. No MinUtxos required. In fact perhaps
-      -- this step can be skipped and we can go straight to prebalancer.
-      unbalancedCollTx :: Transaction
-      unbalancedCollTx = addTxCollateral unbalancedTx' collateral
```

Comment:
This function should NOT be removed.

`addTxCollateral` is still needed for `Nami`.

Please use the same logic as in https://github.com/Plutonomicon/cardano-transaction-lib/pull/455/commits/c559f656f99a10692534cbd5ecc12f7013056993

Reply:
Unfortunately, that code won't work, because I got rid of the `Nami _` case for `Wallet` type; it's all a newtype to `Cip30Wallet` now. We also don't use Nami for submission, preferring Ogmios, AFAIK. Could you give me further details about the logic to implement in the merge?

Reply:
We do need to add this back in, especially if Gero have their own add collateral functionality.